primitives/consensus/babe/src/digests.rs

```diff
@@ -70,7 +71,7 @@ pub struct SecondaryVRFPreDigest {
 /// A BABE pre-runtime digest. This contains all data required to validate a
 /// block and for the BABE runtime module. Slots can be assigned to a primary
 /// (VRF based) and to a secondary (slot number based).
-#[derive(Clone, RuntimeDebug, Encode, Decode)]
+#[derive(Clone, RuntimeDebug, Encode, Decode, MaxEncodedLen, TypeInfo)]
```

Comment:
Is the `TypeInfo` required to be a `pallet::storage` type here? 
https://github.com/paritytech/substrate/blob/7bffe6100a910f53d1c8f68738346dd7b7ddd231/frame/babe/src/lib.rs#L273

Reply:
Yep